libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/AdaptiveDialogBot.cs

```diff
@@ -0,0 +1,103 @@
+
+        private static ConcurrentDictionary<ResourceExplorer, LanguageGeneratorManager> _languageGeneratorManagers = new ConcurrentDictionary<ResourceExplorer, LanguageGeneratorManager>();
+
+        private readonly string _defaultLocale;
+        private readonly string _rootDialogId;
+
+        private readonly ILogger _logger;
+        private readonly ResourceExplorer _resourceExplorer;
+        private readonly IStorage _storage;
+        private readonly BotFrameworkClient _botFrameworkClient;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AdaptiveDialogBot"/> class.
+        /// </summary>
+        /// <param name="configuration">An <see cref="IConfiguration"/> instance.</param>
+        /// <param name="logger">An <see cref="ILogger"/> instance.</param>
+        /// <param name="storage">The <see cref="IStorage"/> implementation to use for this <see cref="Dialog"/>.</param>
+        /// <param name="resourceExplorer">The Bot Builder <see cref="ResourceExplorer"/> to load the <see cref="Dialog"/> from.</param>
+        /// <param name="botFrameworkClient">A <see cref="BotFrameworkClient"/> for making calls to Bot Builder Skills.</param>
+        public AdaptiveDialogBot(
+            IConfiguration configuration,
+            ILogger<AdaptiveDialogBot> logger,
+            ResourceExplorer resourceExplorer,
+            IStorage storage = null,
+            BotFrameworkClient botFrameworkClient = null)
+        {
+            _defaultLocale = configuration.GetSection(DefaultLocaleKey).Value ?? DefaultLocale;
+            _rootDialogId = configuration.GetSection(RootDialogKey).Value ?? "Main.dialog";
+
+            _logger = logger;
+            _resourceExplorer = resourceExplorer;
+            _storage = storage ?? new MemoryStorage();
+            _botFrameworkClient = botFrameworkClient;
+        }
+
+        async Task IBot.OnTurnAsync(ITurnContext turnContext, CancellationToken cancellationToken)
+        {
+            _logger.LogInformation($"IBot.OnTurnContext for dialog '{_rootDialogId}'");
+
+            // Set up the TurnState the Dialog is expecting
+            SetUpTurnState(turnContext);
+
+            // Load the Dialog from the ResourceExplorer
+            var rootDialog = await CreateDialogAsync(cancellationToken).ConfigureAwait(false);
+
+            // Run the Dialog
+            await rootDialog.RunAsync(turnContext, turnContext.TurnState.Get<ConversationState>().CreateProperty<DialogState>("DialogState"), cancellationToken).ConfigureAwait(false);
+
+            // Save any updates that have been made
+            await turnContext.TurnState.Get<ConversationState>().SaveChangesAsync(turnContext, false, cancellationToken).ConfigureAwait(false);
+            await turnContext.TurnState.Get<UserState>().SaveChangesAsync(turnContext, false, cancellationToken).ConfigureAwait(false);
+        }
+
+        private void SetUpTurnState(ITurnContext turnContext)
```

Comment:
We should clearly document in the class summary or the OnTurn method summary what we are putting in turnState (our contract) so devs know what's there and it is easier to discover them.